src/goto-instrument/cover.cpp

```diff
@@ -251,14 +256,59 @@ std::unique_ptr<cover_configt> get_cover_config(
   return cover_config;
 }
 
+/// Build data structures controlling coverage from command-line options.
+/// Include options that depend on the main function specified by the user.
+/// \param options: command-line options
+/// \param main_id: symbol of the user-specified main program function
+/// \param symbol_table: global symbol table
+/// \param message_handler: used to log incorrect option specifications
+/// \return a cover_configt on success, or null otherwise.
+std::unique_ptr<cover_configt> get_cover_config(
+  const optionst &options,
```

Comment:
We are doing C++11 -- so no need for the pointer; simply return the thing, and make sure you've got a working rvalue constructor.

Reply:
Ah, and if you also want to signal error, use `optionalt`, not a null pointer.